lib/govuk_sidekiq/error_handling.rb

```diff
@@ -0,0 +1,33 @@
+module GovukSidekiq
+  module ErrorHandling
+    class RetryWorker < StandardException
+      def initialize(original_exception)
+        @original_exception = original_exception
+      end
+
+      attr_reader :original_exception
+    end
+
+    class Middleware
+      def call(worker, job, queue)
+        yield
+      rescue StandardException => e
+        handle_exception(job, e)
+      end
+
+      def handle_exception(job, e)
+        if last_try?(job)
+          raise e
+        else
+          raise RetryWorker.new(e)
```

Comment:
I'm not actually sure you need to pass the exception through here I think Ruby might auto associate it as the cause. I find the Ruby exception and cause stuff a bit confusing though.